jerry-ext/external/external-register.c

```diff
@@ -0,0 +1,43 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript-ext/external.h"
+
+
+/**
+ * Register a JavaScript function in the global object.
+ *
+ * Note:
+ *      returned value must be freed with jerry_release_value, when it is no longer needed.
+ *
+ * @return true value - if the operation was successful
+ *         value marked with error flag - otherwise
+ */
+jerry_value_t
+jerryx_external_register_global_function (const jerry_char_t *name_p, /**< name of the function */
```

Comment:
Aren't builtins called that way because they are built-in to the core?
I don't see why calling it builtins is a fitting...

I still think splitting the header and naming it after the functions themselves (`jerryscript-ext/gc.h`, `jerryscript-ext/print.h`, `jerryscript-ext/assert.h`, ...) is easiest to understand to the user. I think that's more user-friendly than a non-descript name like `external.h` or `builtin.h`.